controller/connectors/outlet.go

```diff
@@ -54,24 +23,43 @@ type Outlet struct {
 	Reverse   bool   `json:"reverse"`
 }
 
-func (o Outlet) IsValid() error {
+func (o Outlet) outputPin(drivers *drivers.Drivers) (driver.OutputPin, error) {
+	pindriver, err := drivers.Get("rpi")
+	if err != nil {
+		return nil, fmt.Errorf("inlet %s driver lookup failure: %v", o.Name, err)
+	}
+	outputDriver, ok := pindriver.(driver.Output)
+	if !ok {
+		return nil, fmt.Errorf("driver for inlet %s is not an inlet driver", o.Name)
+	}
+	outputPin, err := outputDriver.GetOutputPin(fmt.Sprintf("GP%d", o.Pin))
+	if err != nil {
+		return nil, fmt.Errorf("no valid input pin %d: %v", o.Pin, err)
+	}
+	return outputPin, nil
+}
+
+func (o Outlet) IsValid(drivers *drivers.Drivers) error {
 	if o.Name == "" {
 		return fmt.Errorf("Outlet name can not be empty")
 	}
-	_, ok := ValidGPIOPins[o.Pin]
-	if !ok {
-		return fmt.Errorf("GPIO Pin %d is not valid", o.Pin)
+	if _, err := o.outputPin(drivers); err != nil {
+		return fmt.Errorf("outlet %s did not get associated with a driver pin", o.Name)
```

Comment:
this will confuse us, its masking the original error, better to just return the original error instead of creating new one, since this is part of the create logic caller already know name and pin of the outlet